File: apps/desktop/src/rewrite_css.py
import re

with open('styles.css', 'r') as f:
    css = f.read()

# 1. Remove backdrop-filters
css = re.sub(r'^[ \t]*-webkit-backdrop-filter:.*?;[ \t]*\n', '', css, flags=re.MULTILINE)
css = re.sub(r'^[ \t]*backdrop-filter:.*?;[ \t]*\n', '', css, flags=re.MULTILINE)

# 2. Border-radius
css = re.sub(r'border-radius:\s*999px', r'border-radius: var(--radius-2)', css)
css = re.sub(r'border-radius:\s*(1[2-9]|[2-9][0-9])px', r'border-radius: var(--radius-3)', css)
css = re.sub(r'border-radius:\s*[7-9]px', r'border-radius: var(--radius-2)', css)

# 3. Gradients and colors
css = re.sub(r'var\(--kira-sweep\)', 'var(--accent-strong)', css)
css = re.sub(r'var\(--kira-sweep-soft\)', 'var(--accent-weak)', css)
css = re.sub(r'var\(--kira-tint\)', 'var(--accent-faint)', css)
css = re.sub(r'var\(--kira-halo\)', 'transparent', css)

# Let's remove the variables themselves from the :root just to be clean
# Actually, it's safer to keep them but redefine them as simple colors just in case some inline style uses it
css = re.sub(r'--kira-sweep:\s*linear-gradient[^;]+;', '--kira-sweep: var(--accent-strong);', css)
css = re.sub(r'--kira-sweep-soft:\s*linear-gradient[^;]+;', '--kira-sweep-soft: var(--accent-weak);', css)
css = re.sub(r'--kira-tint:\s*color-mix[^;]+;', '--kira-tint: var(--accent-faint);', css)
css = re.sub(r'--kira-halo:\s*color-mix[^;]+;', '--kira-halo: transparent;', css)

with open('styles.css', 'w') as f:
    f.write(css)

print("Done")

File: apps/desktop/src/rewrite_css.js
const fs = require('fs');

let css = fs.readFileSync('styles.css', 'utf-8');

// 1. Remove all backdrop-filters
css = css.replace(/^[ \t]*-webkit-backdrop-filter:.*?;[ \t]*\n/gm, '');
css = css.replace(/^[ \t]*backdrop-filter:.*?;[ \t]*\n/gm, '');

// 2. Reduce excessive border-radius. We replace 999px, 24px, 18px, 16px, 14px, 12px with a smaller radius.
// Actually, let's look for border-radius: [1-9][0-9]*px and border-radius: 999px
css = css.replace(/border-radius:\s*999px/g, 'border-radius: var(--radius-2)');
css = css.replace(/border-radius:\s*(1[2-9]|[2-9][0-9])px/g, 'border-radius: var(--radius-3)');

// 3. Remove --kira-sweep from variables and replace usages with solid colors.
// Replace var(--kira-sweep) with var(--accent-faint) or something similar if it's an accent, but often it's used for AI buttons.
css = css.replace(/var\(--kira-sweep\)/g, 'var(--accent-strong)');
css = css.replace(/var\(--kira-sweep-soft\)/g, 'var(--accent-weak)');
css = css.replace(/var\(--kira-tint\)/g, 'var(--accent-faint)');
css = css.replace(/var\(--kira-halo\)/g, 'transparent');

// 4. Clean up .kira-dock and .kira-dock.is-open
css = css.replace(/linear-gradient\([^;]*?\)/g, 'var(--surface-inspector)'); 
// wait, replacing all linear gradients might break other things, but Figma doesn't use gradients. Let's do it carefully.

fs.writeFileSync('styles_clean.css', css, 'utf-8');
console.log('Done');
